RecoLocalCalo/EcalRecAlgos/src/EcalUncalibRecHitTimingCCAlgo.cc

```diff
@@ -0,0 +1,167 @@
+#include "RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitTimingCCAlgo.h"
+
+EcalUncalibRecHitTimingCCAlgo::EcalUncalibRecHitTimingCCAlgo(const float startTime, const float stopTime) : _startTime(startTime), _stopTime(stopTime) {}
+
+double EcalUncalibRecHitTimingCCAlgo::computeTimeCC(const EcalDataFrame& dataFrame, const std::vector<double> &amplitudes, const EcalPedestals::Item * aped, const EcalMGPAGainRatio * aGain, const FullSampleVector &fullpulse, EcalUncalibratedRecHit& uncalibRecHit) {
+  const unsigned int nsample = EcalDataFrame::MAXSAMPLES;
+
+  double maxamplitude = -std::numeric_limits<double>::max();
+
+  double pulsenorm = 0.;
+
+  std::vector<double> pedSubSamples(nsample);
+  for(unsigned int iSample = 0; iSample < nsample; iSample++) {
+
+    const EcalMGPASample &sample = dataFrame.sample(iSample);
+
+    double amplitude = 0.;
+    int gainId = sample.gainId();
+
+    double pedestal = 0.;
+    double gainratio = 1.;
+
+    if (gainId==0 || gainId==3) {
+      pedestal = aped->mean_x1;
+      gainratio = aGain->gain6Over1()*aGain->gain12Over6();
+    }
+    else if (gainId==1) {
+      pedestal = aped->mean_x12;
+      gainratio = 1.;
+    }
+    else if (gainId==2) {
+      pedestal = aped->mean_x6;
+      gainratio = aGain->gain12Over6();
+    }
+
+    amplitude = ((double)(sample.adc()) - pedestal) * gainratio;
+
+    if (gainId == 0) {
+      //saturation
+      amplitude = (4095. - pedestal) * gainratio;
+    }
+
+    pedSubSamples.at(iSample) = amplitude;
+
+    if (amplitude>maxamplitude) {
+      maxamplitude = amplitude;
+    }
+    pulsenorm += fullpulse(iSample);
+  }
+
+  std::vector<double>::const_iterator amplit;
+  for(amplit=amplitudes.begin(); amplit<amplitudes.end(); ++amplit) {
+    int ipulse = std::distance(amplitudes.begin(),amplit);
+    int bx = ipulse - 5;
+    int firstsamplet = std::max(0,bx + 3);
+    int offset = 7-3-bx;
```

Comment:
Use named constants instead of magic numbers.

Reply:
Those are magic numbers from here:
https://github.com/cms-sw/cmssw/blob/5b444b50ba85174b79a42b16d4a18c2eb6b653ab/RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitTimeWeightsAlgo.h#L88
Not even sure how to call them

Reply:
Put a comment in the code then and point to the original code where they come from. Maybe at some point someone will take it in him/her to use change everything to named constants properly.